source/gui/__init__.py

```diff
@@ -203,14 +202,8 @@ def onExecuteUpdateCommand(self, evt):
 				updateCheck.executePendingUpdate()
 
 	def evaluateUpdatePendingUpdateMenuItemCommand(self):
-		try:
-			self.sysTrayIcon.menu.Remove(self.sysTrayIcon.installPendingUpdateMenuItem)
-		except:
-			log.debug("Error while removing  pending update menu item", exc_info=True)
-			pass
-		if not globalVars.appArgs.secure and updateCheck and updateCheck.isPendingUpdate():
-			self.sysTrayIcon.menu.Insert(self.sysTrayIcon.installPendingUpdateMenuItemPos,self.sysTrayIcon.installPendingUpdateMenuItem)
-
+		self.sysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand()

```

Comment:
can a deprecation warning be added here? once that is done this PR should be fine to be merged.

```suggestion
		log.warning(
			"MainFrame.evaluateUpdatePendingUpdateMenuItemCommand is deprecated. "
			"Use SysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand instead.",
			stack_info=True,
		)
		self.sysTrayIcon.evaluateUpdatePendingUpdateMenuItemCommand()
		
```

Reply:
@seanbudd,, I have applied your suggestion in ad1f2ad4be8a584af6d76db118074fc78eef6d87. This suggestion is in line with what is done in the rest of NVDA code base. The PR is ready.

Side note: I do not fully understand how deprecation is managed in NVDA code base. When deprecation applies to a module level variable/function, we have the warning only if `NVDAState._allowDeprecatedAPI` returns True. For methods in a class, we get the warning in any case and we do not check `NVDAState._allowDeprecatedAPI`'s return value. Why?

Reply:
thanks.

It is not totally clear on how to handle this for class methods and variables.
A wrapper like this might be possible: https://stackoverflow.com/a/30253848/8030743
We just haven't scheduledtime to investigate a standard syntax to be used for class variables.

Another alternative is similar to existing code:

```py
class Foo
  if NVDAState._allowDeprecatedAPI:
    def example(): pass
```